classes/file_manager.php

```diff
@@ -0,0 +1,186 @@
+     *
+     * @param stored_file $uploadedconfig
+     * @param string $password
+     * @return bool
+     */
+    public function is_file_valid_seb_config(stored_file $uploadedconfig, string $password = '') : bool {
+        $content = $uploadedconfig->get_content();
+        // Attempt to decrypt file contents.
+        $content = seb_cipher::decrypt($content, $password);
+        // Validate file contents, including checking it is an seb config xml file.
+        return $this->validate_seb_contents($content);
+    }
+
+    /**
+     * Create a new file with the same file name and contents of input, in the safe exam browser file area.
+     *
+     * @param stored_file $file
+     * @param string $cmid
+     * @throws \coding_exception
+     * @throws \file_exception
+     * @throws \stored_file_creation_exception
+     */
+    public function save_file_in_module(stored_file $file, string $cmid) {
+        $fs = get_file_storage();
+        if (get_coursemodule_from_id('quiz', $cmid)) { // Validate course module id.
+            $context = \context_module::instance($cmid);
+            if (!$fs->file_exists($context->id, self::COMPONENT, self::FILE_AREA,
+                    $file->get_itemid(), $file->get_filepath(), $file->get_filename())) {
+                $newfile = $fs->create_file_from_storedfile([
+                    'contextid' => $context->id,
+                    'component' => self::COMPONENT,
+                    'filearea' => self::FILE_AREA,
+                ], $file);
+            }
+        }
+    }
+
+    /**
+     * Create a new file with the same file name and contents of input, in the safe exam browser file area.
+     *
+     * @param stored_file $file
+     * @param string $userid
+     * @throws \file_exception
+     * @throws \stored_file_creation_exception
+     */
+    public function save_file_in_user_draft(stored_file $file, string $userid) {
+        $fs = get_file_storage();
+        $context = \context_user::instance($userid);
+        if (!$fs->file_exists($context->id, 'user', 'draft',
+                $file->get_itemid(), $file->get_filepath(), $file->get_filename())) {
+            $newfile = $fs->create_file_from_storedfile([
+                'contextid' => $context->id,
+                'component' => 'user',
+                'filearea' => 'draft',
+            ], $file);
+        }
+    }
+
+    /**
+     * Check that it is a valid seb config file.
+     *
+     * @param string $content
+     * @return bool
+     */
+    private function validate_seb_contents(string $content) : bool {
+        $valid = true;
+        $lines = explode(PHP_EOL, $content);
+
+        // Make sure file length is at least two lines, otherwise exit early.
+        if (count($lines) < 2) {
+            return false;
+        }
+
+        // Check first line declares file as xml.
+        if (strpos($lines[0], '<?xml') !== 0) {
+            $valid = false;
+        }
+
+        // Check second line declares file as plist.
```

Comment:
Why? Can you please add more info as you add some comments. Otherwise it's a bit confusing

Reply:
Added link to SEB docs and explanation in method docs and expanded description for each step.